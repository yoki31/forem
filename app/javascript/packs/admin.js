import { Application } from '@hotwired/stimulus';
import { definitions } from 'stimulus:../admin/controllers'; // eslint-disable-line import/no-unresolved
import Rails from '@rails/ujs';
import 'focus-visible';

// Initialize Rails unobtrusive scripting adapter
// https://github.com/rails/rails/blob/main/actionview/app/assets/javascripts/README.md#es2015
Rails.start();

// This loads all the Stimulus controllers at the same time for the admin
// section of the application.

const application = Application.start();
application.load(definitions);
