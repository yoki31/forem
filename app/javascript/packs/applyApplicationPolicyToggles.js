import { getUserDataAndCsrfToken } from '@utilities/getUserDataAndCsrfToken';
/**
 * Responsible for hiding or showing elements that match each of the given user
 * policies.  While this function is "oblivious" to what it's hiding, it
 * coordinates between the rendered HTML and the user data to show or hide
 * elements that present functionality available or not available to the given
 * user.
 *
 * A critical assumption is that we are not employing "security through
 * obscurity".  That is to say, if we accidentally show the link, the server
 * will enforce the correct policy.
 */
getUserDataAndCsrfToken().then(({ currentUser }) => {
  if (currentUser.policies) {
    currentUser.policies.forEach((policy) => {
      const elements = document.getElementsByClassName(policy.dom_class);
      for (const element of elements) {
        if (policy.visible) {
          element.classList.remove('hidden');
        } else {
          element.classList.add('hidden');
        }
      }
    });
  }
});
