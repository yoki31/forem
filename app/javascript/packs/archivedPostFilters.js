function initializeArchivedPostFilter() {
  const link = document.getElementById('toggleArchivedLink');
  if (link) {
    link.addEventListener('click', toggleArchivedPosts);
  }
}

function archivedPosts() {
  return document.getElementsByClassName('story-archived');
}

function showArchivedPosts() {
  const posts = archivedPosts();

  for (let i = 0; i < posts.length; i += 1) {
    posts[i].classList.remove('hidden');
  }
}

function hideArchivedPosts() {
  const posts = archivedPosts();

  for (let i = 0; i < posts.length; i += 1) {
    posts[i].classList.add('hidden');
  }
}

function toggleArchivedPosts(e) {
  e.preventDefault();
  const link = e.target;

  if (link.innerHTML.match(/Show/)) {
    link.innerHTML = 'Hide archived';
    showArchivedPosts();
  } else {
    link.innerHTML = 'Show archived';
    hideArchivedPosts();
  }
}

initializeArchivedPostFilter();
