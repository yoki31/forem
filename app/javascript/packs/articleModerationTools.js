/* global userData */
import { isModerationPage } from '@utilities/moderation';

getCsrfToken().then(() => {
  const articleContainer = document.getElementById('article-show-container');

  if (articleContainer) {
    const user = userData();
    const { authorId: articleAuthorId, path } = articleContainer.dataset;

    const initializeModerationsTools = async () => {
      const { initializeActionsPanel } = await import(
        '../actionsPanel/initializeActionsPanelToggle'
      );

      // If the user can moderate an article give them access to this panel.
      // Note: this assumes we're within the article context (which is the case
      // given the articleContainer)
      const canModerateArticles = user?.policies?.find(
        (o) =>
          o.dom_class === 'js-policy-article-moderate' && o.visible === true,
      );

      // article show page
      if (canModerateArticles) {
        // <2022-05-09 Mon> [@jeremyf] the user.id is an integer and
        // articleAuthorId is a string so our logic is such that we always
        // initializeActionsPanel and initializeFlagUserModal; I'm asking
        // product to clarify if we want mods to boost their own posts.
        if (user?.id !== articleAuthorId && !isModerationPage()) {
          initializeActionsPanel(user, path);
          // "/mod" page
        } else if (isModerationPage()) {
          initializeActionsPanel(user, path);
        }
      }
    };

    initializeModerationsTools();
  }
});
