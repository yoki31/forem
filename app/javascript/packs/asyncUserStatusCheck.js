import '@utilities/document_ready';

export async function asyncUserStatusCheck() {
  const profile = document.querySelector('.profile-header__details');

  if (profile && !profile.dataset.statusChecked) {
    await window
      .fetch(profile.dataset.url)
      .then((res) => res.json())
      .then((data) => {
        profile.dataset.statusChecked = true;
        const { suspended, spam } = data;

        let status = '';
        if (spam) status = 'Spam';
        else if (suspended) status = 'Suspended';

        if (status) {
          const indicator = `<span data-testid="user-status" class="ml-3 c-indicator c-indicator--danger c-indicator--relaxed">${status}</span>`;
          profile.querySelector('.js-username-container').innerHTML += indicator;
        }
      });
  }
}






document.ready.then(() => {
  asyncUserStatusCheck();
});
