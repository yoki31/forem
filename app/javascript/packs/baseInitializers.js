import { initializeCommentDate } from './initializers/initializeCommentDate';
import { initializeCommentPreview } from './initializers/initializeCommentPreview';
import { initializeTimeFixer } from './initializers/initializeTimeFixer';
import { initializeNotifications } from './initializers/initializeNotifications';
import { initializeDateHelpers } from './initializers/initializeDateTimeHelpers';
import { initializeSettings } from './initializers/initializeSettings';
import {
  showUserAlertModal,
  showModalAfterError,
} from '@utilities/showUserAlertModal';

initializeCommentDate();
initializeCommentPreview();
initializeSettings();
initializeNotifications();
initializeTimeFixer();
initializeDateHelpers();

InstantClick.on('change', () => {
  initializeCommentDate();
  initializeCommentPreview();
  initializeSettings();
  initializeNotifications();
});

window.showUserAlertModal = showUserAlertModal;
window.showModalAfterError = showModalAfterError;
