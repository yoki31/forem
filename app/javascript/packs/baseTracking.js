/*eslint-disable prefer-rest-params*/
/* global isTouchDevice */

function initializeBaseTracking() {
  showCookieConsentBanner();
  trackGoogleAnalytics3();
  trackGoogleAnalytics4();
  trackCustomImpressions();
  trackEmailClicks();
}

// Google Anlytics 3 is deprecated, and mostly not supported, but some sites may still be using it for now.
function trackGoogleAnalytics3() {
  let wait = 0;
  let addedGA = false;
  const gaTrackingCode = document.body.dataset.gaTracking;
  if (gaTrackingCode && localStorage.getItem('cookie_status') === 'allowed') {
    const waitingOnGA = setInterval(() => {
      if (!addedGA) {
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
          (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
                                  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
      }
      addedGA = true;
      wait++;
      if (window.ga && ga.create) {
        ga('create', gaTrackingCode, 'auto');
        ga('set', 'anonymizeIp', true);
        ga('send', 'pageview', location.pathname + location.search);
        clearInterval(waitingOnGA);
      }
      if (wait > 85) {
        clearInterval(waitingOnGA);
        fallbackActivityRecording();
      }
    }, 25);
    eventListening();
  } else if (gaTrackingCode) {
    fallbackActivityRecording();
  }
}

function trackGoogleAnalytics4() {
  let wait = 0;
  let addedGA4 = false;
  const ga4MeasurementCode = document.body.dataset.ga4TrackingId;
  if (ga4MeasurementCode) {
    const waitingOnGA4 = setInterval(() => {
      if (!addedGA4) {
        //Dynamically add the Google Analytics 4 script tag
        const script = document.createElement('script');
        script.src = `//www.googletagmanager.com/gtag/js?id=${  ga4MeasurementCode}`;
        script.async = true;
        document.head.appendChild(script);
      }
      addedGA4 = true;
      wait++;
      if (window.google_tag_manager) {
        //Define the gtag function and call it. Adapted from https://stackoverflow.com/questions/22716542/google-analytics-code-explanation %>
        window.dataLayer = window.dataLayer || [];
        // eslint-disable-next-line no-inner-declarations
        function gtag(){window.dataLayer.push(arguments);}

        window['gtag'] = window['gtag'] || function () {
          window.dataLayer.push(arguments)
        }
        const consent = localStorage.getItem('cookie_status') === 'allowed' ? 'granted' : 'denied';
        gtag('js', new Date());
        gtag('config', ga4MeasurementCode, { 'anonymize_ip': true });
        gtag('consent', 'default', {
          'ad_storage': consent,
          'analytics_storage': consent
        });
        clearInterval(waitingOnGA4);
      }
      if (wait > 85) {
        clearInterval(waitingOnGA4);
        fallbackActivityRecording();
      }
    }, 25);
    eventListening();
  }
}


function fallbackActivityRecording() {
  const tokenMeta = document.querySelector("meta[name='csrf-token']")
  if (!tokenMeta) {
    return
  }
  const csrfToken = tokenMeta.getAttribute('content')
  const w = Math.max(document.documentElement.clientWidth, window.innerWidth || 0);
  const h = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
  const screenW = window.screen.availWidth;
  const screenH = window.screen.availHeight;
  const dataBody = {
    path: location.pathname + location.search,
    user_language: navigator.language,
    referrer: document.referrer,
    user_agent: navigator.userAgent,
    viewport_size: `${h}x${w}`,
    screen_resolution: `${screenH}x${screenW}`,
    document_title: document.title,
    document_encoding: document.characterSet,
    document_path: location.pathname + location.search,
  };
  window.fetch('/fallback_activity_recorder', {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'X-CSRF-Token': csrfToken,
    },
    body: JSON.stringify(dataBody),
    credentials: 'same-origin'
  });
}

function eventListening(){
  const registerNowButt = document.getElementById("cta-comment-register-now-link");
  if (registerNowButt) {
    registerNowButt.onclick = function(){
      ga('send', 'event', 'click', 'register-now-click', null, null);
    }
  }
}

// eslint-disable-next-line no-unused-vars
function ga4eventListening(){
  const registerNowButt = document.getElementById("cta-comment-register-now-link");
  if (registerNowButt) {
    registerNowButt.onclick = function(){
      gtag('event', 'register-now-click' );
    }
  }
}

function checkUserLoggedIn() {
  const {body} = document;
  if (!body) {
    return false;
  }

  return body.getAttribute('data-user-status') === 'logged-in';
}

function trackCustomImpressions() {
  setTimeout(()=> {
    const ArticleElement = document.getElementById('article-body') || document.getElementById('comment-article-indicator');
    const tokenMeta = document.querySelector("meta[name='csrf-token']")
    const isBot = /bot|google|baidu|bing|msn|duckduckbot|teoma|slurp|yandex/i.test(navigator.userAgent) // is crawler
    // eslint-disable-next-line no-unused-vars
    const windowBigEnough =  window.innerWidth > 1023

    // page view
    if (ArticleElement && tokenMeta && !isBot) {
      // See https://github.com/forem/forem/blob/main/app/controllers/page_views_controller.rb
      //
      // If you change the 10, you should look at the PageViewsController as well.
      const randomNumber = Math.floor(Math.random() * 10); // 1 in 10; Only track 1 in 10 impressions
      if (!checkUserLoggedIn() && randomNumber != 1) {
        return;
      }
      const dataBody = {
        article_id: ArticleElement.dataset.articleId,
        referrer: document.referrer,
        user_agent: navigator.userAgent,
      };
      const csrfToken = tokenMeta.getAttribute('content');
      trackPageView(dataBody, csrfToken);
      let timeOnSiteCounter = 0;
      const timeOnSiteInterval = setInterval(()=> {
        timeOnSiteCounter++
        const ArticleElement = document.getElementById('article-body') || document.getElementById('comment-article-indicator');
        if (ArticleElement && checkUserLoggedIn()) {
          trackFifteenSecondsOnPage(ArticleElement.dataset.articleId, csrfToken);
        } else {
          clearInterval(timeOnSiteInterval);
        }
        if ( timeOnSiteCounter > 118 ) {
          clearInterval(timeOnSiteInterval);
        }
      }, 15000)
    }

  }, 1800)
}

function trackEmailClicks() {
  const urlParams = new URLSearchParams(window.location.search);

    if (urlParams.get('ahoy_click') === 'true' && urlParams.get('t') && urlParams.get('s') && urlParams.get('u')){
      const dataBody = {
        t: urlParams.get('t'),
        c: urlParams.get('c'),
        u: decodeURIComponent(urlParams.get('u')),
        s: urlParams.get('s'),
        bb: urlParams.get('bb'),
      };
      window.fetch('/ahoy/email_clicks', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(dataBody),
        credentials: 'same-origin'
      });
      // Remove t,c,u,s params and ahoy_click param from url without modifying the history
      urlParams.delete('t');
      urlParams.delete('c');
      urlParams.delete('u');
      urlParams.delete('s');
      urlParams.delete('ahoy_click');
      urlParams.delete('bb');
      const newUrl = `${window.location.pathname  }?${  urlParams.toString()}`;
      window.history.replaceState({}, null, newUrl);
    }
}

function showCookieConsentBanner() {
  // if current url includes ?cookietest=true
  if (shouldShowCookieBanner()) {
    // show modal with cookie consent
    const cookieDiv = document.getElementById('cookie-consent');

    if (cookieDiv && localStorage.getItem('cookie_status') !== 'allowed' && localStorage.getItem('cookie_status') !== 'dismissed') {
      cookieDiv.innerHTML = `
        <div class="cookie-consent-modal">
          <div class="cookie-consent-modal__content">
            <p>
              <strong>Some content on our site requires cookies for personalization.</strong>
            </p>
            <p>
              Read our full <a href="/privacy">privacy policy</a> to learn more.
            </p>
            <div class="cookie-consent-modal__actions">
              <button class="c-btn c-btn--secondary" id="cookie-dismiss">
                Dismiss
              </button>
              <button class="c-btn c-btn--primary" id="cookie-accept">
                Accept Cookies
              </button>
            </div
          </div>
        </div>
      `;

      document.getElementById('cookie-accept').onclick = (() => {
        localStorage.setItem('cookie_status', 'allowed');
        cookieDiv.style.display = 'none';
        if (window.gtag) {
          gtag('consent', 'update', {
            'ad_storage': 'granted',
            'analytics_storage': 'granted'
          });
        }
      });

      document.getElementById('cookie-dismiss').onclick = (() => {
        localStorage.setItem('cookie_status', 'dismissed');
        cookieDiv.style.display = 'none';
      });
    }
  }
}

function shouldShowCookieBanner() {
  const { userStatus, cookieBannerUserContext, cookieBannerPlatformContext } = document.body.dataset;
  function determineActualPlatformContext() {
    if (navigator.userAgent.includes('DEV-Native')) {
      return 'mobile_app'
    } else if (isTouchDevice()) {
      return 'mobile_web'
    }
    return 'desktop_web'
  }

  // Determine the actual platform context
  const actualPlatformContext = determineActualPlatformContext();

  // Check if either user or platform context is set to 'off'
  if (cookieBannerUserContext === 'off' || cookieBannerPlatformContext === 'off') {
    return false;
  }

  // Check based on user status
  const showForUserContext = (userStatus === 'logged-in' && cookieBannerUserContext === 'all') ||
                             (userStatus !== 'logged-in' && cookieBannerUserContext !== 'off');

  // Check based on platform context
  const showForPlatformContext = (cookieBannerPlatformContext === 'all') ||
                                 (cookieBannerPlatformContext === 'all_web' && ['desktop_web', 'mobile_web'].includes(actualPlatformContext)) ||
                                 (cookieBannerPlatformContext === actualPlatformContext);

  // Return true if both user context and platform context conditions are met
  return showForUserContext && showForPlatformContext;
}

function trackPageView(dataBody, csrfToken) {
  window.fetch('/page_views', {
    method: 'POST',
    headers: {
      'X-CSRF-Token': csrfToken,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(dataBody),
    credentials: 'same-origin',
  })
}

function trackFifteenSecondsOnPage(articleId, csrfToken) {
  window.fetch(`/page_views/${  articleId}`, {
    method: 'PATCH',
    headers: {
      'X-CSRF-Token': csrfToken,
      'Content-Type': 'application/json',
    },
    credentials: 'same-origin',
  }).catch((error) => console.error(error))
}

window.InstantClick.on('change', () => {
  initializeBaseTracking();
});
initializeBaseTracking();
