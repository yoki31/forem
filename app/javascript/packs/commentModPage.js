import { updateExperienceLevel } from '../actionsPanel/actionsPanel';

/**
 * A thumbsup reaction on a comment/article will invalidate a previous thumbsdown
 * or vomit reaction (they will be deleted on the server by the reaction handler)
 * and vice versa. This function updates the UI to match.
 * @param {HTMLButtonElement} clickedBtn The reaction button that was clicked
 */
function toggleContradictoryReactions(clickedBtn) {
  const contentActions = document.querySelector('#content-mod-actions');

  if (clickedBtn.parentElement === contentActions) {
    const upVote = contentActions.querySelector("[data-category='thumbsup']");
    const downVote = contentActions.querySelector(
      "[data-category='thumbsdown']",
    );
    const vomitVote = contentActions.querySelector("[data-category='vomit']");

    if (clickedBtn.dataset.category === 'thumbsup') {
      downVote.classList.remove('reacted');
      vomitVote.classList.remove('reacted');
    } else {
      upVote.classList.remove('reacted');
    }
  }
}

async function updateMainReactions(clickedBtn) {
  const { reactableType, category, reactableId } = clickedBtn.dataset;
  try {
    const response = await fetch('/reactions', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'X-CSRF-Token': window.csrfToken,
        'Content-Type': 'application/json',
      },
      credentials: 'same-origin',
      body: JSON.stringify({
        reactable_type: reactableType,
        category,
        reactable_id: reactableId,
      }),
    });

    const outcome = await response.json();

    if (outcome.result === 'create') {
      toggleContradictoryReactions(clickedBtn);
      clickedBtn.classList.add('reacted');
    } else if (outcome.result === 'destroy') {
      clickedBtn.classList.remove('reacted');
    } else {
      // eslint-disable-next-line no-alert
      alert(`Error: ${outcome.error}`);
      // eslint-disable-next-line no-console
      console.error(`Error: ${outcome.error}`);
    }
  } catch (error) {
    // eslint-disable-next-line no-alert
    alert(`Error: ${error}`);
  }
}

// Experience-Level JS
Array.from(document.getElementsByClassName('level-rating-button')).forEach(
  (btn) => {
    btn.addEventListener('click', () => {
      updateExperienceLevel(
        btn.dataset.userId,
        btn.dataset.articleId,
        btn.value,
        btn.dataset.group,
      );
    });
  },
);

document
  .querySelectorAll('.reaction-button, .reaction-vomit-button')
  .forEach((btn) => {
    btn.addEventListener('click', async () => {
      await updateMainReactions(btn);
    });
  });

const form = document.getElementsByClassName('button_to')[0];
if (form) {
  form.addEventListener('submit', (e) => {
    e.preventDefault();
    if (confirm('Are you SURE you want to delete this comment?')) {
      form.submit();
    }
  });
}
