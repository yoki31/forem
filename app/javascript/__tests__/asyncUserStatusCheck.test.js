import fetch from 'jest-fetch-mock';
import '@testing-library/jest-dom';

import { asyncUserStatusCheck } from '../packs/asyncUserStatusCheck';

global.fetch = fetch;

function fakeUserIsSuspended() {
  return JSON.stringify({
    id: 123,
    username: 'user123',
    suspended: true,
  });
}

function fakeUserNotSuspended() {
  return JSON.stringify({
    id: 123,
    username: 'user123',
  });
}

describe('asyncUserStatusCheck', () => {
  beforeEach(() => {
    global.Honeybadger = { notify: jest.fn() };

    fetch.resetMocks();

    window.document.body.innerHTML = `
    <div class="profile-header__details" data-url="/users/123">
      <div class="js-username-container">
        <h1 class="crayons-title">User Hasaname</h1>
      </div>
      <p class="">Animi et qui. Voluptatum voluptas omnis. Libero voluptatem cum. Unde.</p>

      <div class="profile-header__meta">
      </div>
    </div>
    `;
  });

  describe('user is **NOT** suspended', () => {
    beforeEach(() => {
      fetch.mockResponse(fakeUserNotSuspended());
    });

    it('uses fetch to retrieve data async', async () => {
      await asyncUserStatusCheck();
      expect(fetch).toHaveBeenCalledWith('/users/123');
    });

    it('does not badge the user', async () => {
      await asyncUserStatusCheck();
      expect(document.body.innerHTML).toMatchSnapshot();
      expect(document.body.innerHTML).not.toMatch(/Suspended/);
    });
  });

  describe('user **is** suspended', () => {
    beforeEach(() => {
      fetch.mockResponse(fakeUserIsSuspended());
    });

    it('uses fetch to retrieve data async', async () => {
      await asyncUserStatusCheck();
      expect(fetch).toHaveBeenCalledWith('/users/123');
    });

    it('**does** badge the user', async () => {
      await asyncUserStatusCheck();
      expect(document.body.innerHTML).toMatchSnapshot();
      expect(document.body.innerHTML).toMatch(/Suspended/);
    });
  });
});
