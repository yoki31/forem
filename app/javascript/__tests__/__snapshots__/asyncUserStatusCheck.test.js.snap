// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asyncUserStatusCheck user **is** suspended **does** badge the user 1`] = `
"
    <div class=\\"profile-header__details\\" data-url=\\"/users/123\\" data-status-checked=\\"true\\">
      <div class=\\"js-username-container\\">
        <h1 class=\\"crayons-title\\">User Hasaname</h1>
      <span data-testid=\\"user-status\\" class=\\"ml-3 c-indicator c-indicator--danger c-indicator--relaxed\\">Suspended</span></div>
      <p class=\\"\\">Animi et qui. Voluptatum voluptas omnis. Libero voluptatem cum. Unde.</p>

      <div class=\\"profile-header__meta\\">
      </div>
    </div>
    "
`;

exports[`asyncUserStatusCheck user is **NOT** suspended does not badge the user 1`] = `
"
    <div class=\\"profile-header__details\\" data-url=\\"/users/123\\" data-status-checked=\\"true\\">
      <div class=\\"js-username-container\\">
        <h1 class=\\"crayons-title\\">User Hasaname</h1>
      </div>
      <p class=\\"\\">Animi et qui. Voluptatum voluptas omnis. Libero voluptatem cum. Unde.</p>

      <div class=\\"profile-header__meta\\">
      </div>
    </div>
    "
`;
